src/super_gradients/training/models/classification_models/mobilenetv2.py

```diff
@@ -151,17 +148,17 @@ def __init__(
 
     def forward(self, x):
         x = self.features(x)
-        x = x.mean(3).mean(2)
-        x = self.classifier(x)
-        return x
+        if self.backbone_mode:
+            return x
+        else:
+            x = x.mean(3).mean(2)
+            return self.classifier(x)
 
     def _extract_connection_layers_input_channel_size(self):
         """
         Extracts the number of channels out when using mobilenetV2 as yolo backbone
         """
-        curr_layer_input = torch.rand(
-            1, self.in_channels, 320, 320
-        )  # input dims are used to extract number of channels
+        curr_layer_input = torch.rand(1, self.in_channels, 320, 320)  # input dims are used to extract number of channels
```

Comment:
Following my previous comment, do we still need this method?

Reply:
Will be happy to remove it later, looks bad

Reply:
Please add a @TODO comment so wed remember to do so (hopefully).